dbms/src/Server/Server.cpp

```diff
@@ -1546,6 +1545,18 @@ int Server::main(const std::vector<std::string> & /*args*/)
         }
     });
 
+    auto & tmt_context = global_context->getTMTContext();
+    // Resource Control.
+#ifdef DBMS_PUBLIC_GTEST
+    LOG_WARNING(
+        log,
+        "Compiled with ENABLE_TESTS enabled, will use MockLocalAdmissionController, which means tiflash "
+        "ResourceControl will be disabled.");
```

Comment:
Why is this needed?
What's the problem with getting rid of it?

Reply:
If you compile tiflash server using `cmake .. -DENABLE_TESTS=ON`, `DBMS_PUBLIC_GTESTS` will be defined. So `MockLocalAdmissionController` will be used by `RCQ` instead of the real `LAC`.

`MockLAC` will not talk to `GAC` and it return same priority for all pipeline tasks, this is to remaind this behavior.

Reply:
Does `DB::LocalAdmissionController::global_instance = std::make_unique<DB::MockLocalAdmissionController>();` need to be call here?

Reply:
It's called [here](https://github.com/pingcap/tiflash/pull/7927/files/25cb1a0ccf60667dd2e2e284a8ba75eabc1661c0#diff-7688037759fb8f57bcf85cdcb13657390b20b0cf711f8c15ebba7773e529f4d3R494). Because gtest will not compile `Server.cpp`.

Reply:
If Server will not be compiled on DBMS_PUBLIC_GTEST, is it possible to remove this code?

Reply:
Yes. It's ok to remove this log if Server will not be compiled on DBMS_PUBLIC_GTEST.
But how can we disable this behavior? static_assert?

Reply:
Maybe it's not necessary to consider.